src/main/java/io/seqera/migtool/MigTool.java

```diff
@@ -509,4 +487,64 @@ public Object doCall() {
         }
     }
 
+    public static class Builder {
+        private String driverName;
+        private Driver driver;
+        private String url;
+        private String user;
+        private String password;
+        private String dialectName;
+        private Dialect dialect;
+        private String locations;
+        private ClassLoader classLoader;
+        private Pattern pattern;
+
+        public Builder withDriver(String driverName) {
+            this.driverName = driverName;
+            return this;
+        }
+
+        public Builder withUrl(String url) {
+            this.url = url;
+            return this;
+        }
+
+        public Builder withUser(String user) {
+            this.user = user;
+            return this;
+        }
+
+        public Builder withPassword(String password) {
+            this.password = password;
+            return this;
+        }
+
+        public Builder withDialect(String dialectName) {
+            this.dialectName = dialectName;
+            return this;
+        }
+
+        public Builder withLocations(String locations) {
+            this.locations = locations;
+            return this;
+        }
+
+        public Builder withClassLoader(ClassLoader loader) {
+            this.classLoader = loader;
+            return this;
+        }
+
+        public Builder withPattern(String pattern) {
+            if(pattern != null && !pattern.isEmpty())
+                this.pattern = Pattern.compile(pattern);
+            return this;
+        }
+
+        public MigTool build() {
+            this.driver = driverName != null ? Driver.fromDriverName(driverName) : Driver.fromUrl(url);
+            this.dialect = dialectName != null ? Dialect.fromDialectName(dialectName) : Dialect.fromUrl(url);
+            return new MigTool(this);
+        }
+    }
+
```

Comment:
Bit useless. It was avoid on purpose to not implement the builder pattern because it's just redundant